shell/edit/provisioning.cattle.io.cluster/AddOnConfig.vue

```diff
@@ -0,0 +1,132 @@
+<script>
+import { Banner } from '@components/Banner';
+
+import Questions from '@shell/components/Questions';
+import YamlEditor from '@shell/components/YamlEditor';
+import { camelToTitle } from '@shell/utils/string';
+
+export default {
+  components: {
+    Banner,
+    Questions,
+    YamlEditor
+  },
+
+  props: {
+    mode: {
+      type:     String,
+      required: true,
+    },
+
+    value: {
+      type:     Object,
+      required: true,
+    },
+    isEdit: {
```

Comment:
Adding this as a required prop seems excessive to me when `mode` is also passed in. I think it would be preferable to reduce the number of props required to use this component by importing the `_EDIT` constant from `@shell/config/query-params` and comparing that to `mode` within this component (this applies to the Networking component as well)